### PR TITLE
Add support for terminal hyperlinks

### LINKED ISSUE #812: Clickable terminal links
**Describe the feature or problem you'd like to solve**

We could make `issue list` and `mr list` clickable.

**Propose a Solution**

```
!123  Add support for links             (main) <- (links)
```

- Click `!123` to go to the issue or merge request: 
- Click `(links)` to go to the branch. Example: https://example.com/user/repo/-/tree/links

**Additional context**

- [Hyperlinks (a.k.a. HTML-like anchors) in terminal emulators](https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda)

### PR DESCRIPTION
**Pull request by hoelzro**
**Description**
Adds support for OSC 8 terminal links

**Related Issue**

Resolves #812

**How Has This Been Tested?**

I conducted manual tests for this

**Types of changes**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation
- [ ] Chore (Related to CI or Packaging to platforms)

### CODE REVIEW

**Comment by hoelzro** [comment]
I submitted this early for quicker feedback on the code - I'm not a huge fan of the `if displayHyperlinks { return makeHyperlink(...) } else { return ... }`, but I wanted to get something pushed for others to look at.  Would it be overkill to have `MakeHyperlink` just return the display string if hyperlinks are _not_ wanted, so display code making use of hyperlinks doesn't need to dispatch on `DisplayHyperlinks` itself?

Oh, and I'm not attached to the method names I added - they should probably stand to be improved :smile:

**Review by profclems — COMMENTED** [review]
Great work on this @hoelzro!

This is huge. I will take time this evening (my time 😄) to review

**Comment by hoelzro** [comment]
@profclems Awesome, thanks! I just added docs and fixed the tableprinter tests, so it should be ready to roll if it looks good to you!

**Comment by hoelzro** [comment]
@profclems Have you had a chance to look at this?  I know it's a pretty big PR - let me know if it would help to break up a bit! :sweat_smile:

File: pkg/iostreams/iostreams.go [hunk 1]
```diff
@@ -200,6 +203,32 @@ func (s *IOStreams) BackgroundColor() string {
...
+func (s *IOStreams) SetDisplayHyperlinks(displayHyperlinks string) {
+	s.displayHyperlinks = displayHyperlinks
+}
+
+func (s *IOStreams) shouldDisplayHyperlinks() bool {
+	switch s.displayHyperlinks {
+	case "always":
+		return true
+	case "auto":
+		return s.IsaTTY
```
Comment (profclems) [inline pkg/iostreams/iostreams.go]: The OSC doesn't seem to work well when piped to pager. This is what happens:
```sh
Showing 1 open issue in profclems/glab that match your search (Page 1)

8;;https://gitlab.com/profclems/glab/-/issues/278#2788;;  .gitconfig credential helper path does not work on Windows  (gh-cloned)  about 5 months ago
```


Maybe we should check if the result is piped to pager
```go
return s.IsaTTY && (s.pagerCommand == "" || s.pagerCommand == "cat")
```

**Comment by hoelzro** [comment]
@profclems Good catch! I just added a commit to fix that

**Review by profclems — APPROVED** [review]
This is great!

**Pull request merged**